lib/cdo/i18n_backend.rb

```diff
@@ -3,6 +3,62 @@
 require 'cdo/key_value'
 
 module Cdo
+  # TODO: (elijah) enable this for the backend as a whole, and remove the
+  # pathwork implementation in blockly.rb.
+  #
+  # See https://github.com/code-dot-org/code-dot-org/blob/14bc4168b8526a65bccf3780fa6a2822bff02c60/lib/cdo/i18n_backend.rb
+  module I18nSmartTranslate
+    # if called with the :smart option, attempt to improve the given options
+    def translate(locale, key, options = I18n::EMPTY_HASH)
```

Comment:
is this method (and therefore `I18nSmartTranslate.get_smart_translate_options`) used anywhere yet? i'm only seeing `I18nSmartTranslate.get_valid_separator` in this PR

Reply:
Not yet! The idea is that once I add this to the sitewide I18n module, this will be used automatically when we call `I18n.t`